### PR TITLE
Add bg bot announce is customizable

### PR DESCRIPTION
**Pull request by gajet5**
1. Flood protect
2. This option is more like a debag.

### Proof

- None

### Issues

- None

### How2Test

- None

### Todo / Checklist

- [X] None

### CODE REVIEW

File: src/mangosd/mangosd.conf.dist.in [hunk 1]
```diff
@@ -70,8 +70,8 @@ ConfVersion=2010100901
 ###################################################################################################################
 
 RealmID = 1
-DataDir = "."
-LogsDir = ""
+DataDir = "/wow/data"
+LogsDir = "/wow/logs"
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:74]: You shouldn't be changing the default configuration values.

File: src/mangosd/mangosd.conf.dist.in [hunk 2]
```diff
@@ -243,7 +243,7 @@ BindIP = "0.0.0.0"
 UseProcessors = 0
 ProcessPriority = 1
 Compression = 1
-PlayerLimit = 100
+PlayerLimit = 500
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:246]: Same here.

File: src/mangosd/mangosd.conf.dist.in [hunk 3]
```diff
@@ -258,8 +258,8 @@ ChangeWeatherInterval = 600000
 PlayerSave.Interval = 900000
 PlayerSave.Stats.MinLevel = 0
 PlayerSave.Stats.SaveOnlyOnLogout = 1
-Terrain.Preload.Continents = 0
-Terrain.Preload.Instances  = 0
+Terrain.Preload.Continents = 1
+Terrain.Preload.Instances  = 1
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:262]: And here.

File: src/mangosd/mangosd.conf.dist.in [hunk 4]
```diff
@@ -630,11 +630,11 @@ DebuffLimit = 0
 #
 ###################################################################################################################
 
-LogSQL = 1
+LogSQL = 0
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:633]: And here.

File: src/mangosd/mangosd.conf.dist.in [hunk 5]
```diff
@@ -630,11 +630,11 @@ DebuffLimit = 0
...
 PidFile = ""
-LogLevel.Console = 2
-LogLevel.File = 2
-LogLevel.DB = 2
+LogLevel.Console = 1
+LogLevel.File = 1
+LogLevel.DB = 0
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:637]: And here.

File: src/mangosd/mangosd.conf.dist.in [hunk 6]
```diff
@@ -662,7 +662,7 @@ LogFile.Honor = ""
 LogFile.Ra = "Ra.log"
 LogFile.DBError = "DBErrors.log"
 LogFile.DBErrorFix = ""
-LogFile.Loot = "Loot.log"
+LogFile.Loot = ""
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:665]: And here.

File: src/mangosd/mangosd.conf.dist.in [hunk 7]
```diff
@@ -684,7 +684,7 @@ LogsDB.Transactions         = 0
 LogsDB.Battlegrounds        = 0
 Smartlog.Death              = 0
 Smartlog.LongCombat         = 0
-Smartlog.LongCombatDuration = 1800
+Smartlog.LongCombatDuration = 0
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:687]: And here.

File: src/mangosd/mangosd.conf.dist.in [hunk 8]
```diff
@@ -1064,9 +1064,9 @@ RealmZone = 1
 TimeZoneOffset = 0
 DBC.Locale = 255
 DeclinedNames = 0
-StrictPlayerNames = 0
-StrictCharterNames = 0
-StrictPetNames = 0
+StrictPlayerNames = 1
+StrictCharterNames = 1
+StrictPetNames = 1
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:1069]: And here.

File: src/mangosd/mangosd.conf.dist.in [hunk 9]
```diff
@@ -1166,7 +1166,7 @@ GM.JoinOppositeFactionChannels = 0
 AllowTwoSide.Interaction.Group = 0
 AllowTwoSide.Interaction.Guild = 0
 AllowTwoSide.Interaction.Trade = 0
-AllowTwoSide.Interaction.Auction = 0
+AllowTwoSide.Interaction.Auction = 1
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:1169]: And here.

File: src/mangosd/mangosd.conf.dist.in [hunk 10]
```diff
@@ -2850,12 +2850,12 @@ CharDelete.KeepDays = 30
 #        Default: 50
 #
 ###################################################################################################################
-AHBot.Enable = 0
+AHBot.Enable = 1
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:2853]: And here.

File: src/mangosd/mangosd.conf.dist.in [hunk 11]
```diff
@@ -2850,12 +2850,12 @@ CharDelete.KeepDays = 30
...
 AHBot.ah.id = 7
 AHBot.ah.fid = 120
-AHBot.itemcount = 50
-#AHBot.bot.guid = 1123
-#AHBot.bot.account = 32377
+AHBot.itemcount = 3610
+#AHBot.bot.guid = 7
+#AHBot.bot.account = 5
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:2858]: And here.

File: src/mangosd/mangosd.conf.dist.in [hunk 12]
```diff
@@ -2941,10 +2946,11 @@ PlayerBot.ShowInWhoList = 0
 PartyBot.MaxBots = 0
 PartyBot.SkipChecks = 0
 PartyBot.AutoEquip = 1
-PartyBot.RandomGearLevelDifference = 10
+PartyBot.RandomGearLevelDifference = 5
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:2949]: And here.

File: src/mangosd/mangosd.conf.dist.in [hunk 13]
```diff
@@ -2941,10 +2946,11 @@ PlayerBot.ShowInWhoList = 0
...
 
 BattleBot.AutoEquip = 1
-BattleBot.AutoJoin = 0
+BattleBot.AutoJoin = 1
```
Comment (mserajnik) [inline src/mangosd/mangosd.conf.dist.in:2952]: And here.

File: src/realmd/realmd.conf.dist.in [hunk 14]
```diff
@@ -177,7 +177,7 @@ WrongPass.BanType = 0
 ReqEmailVerification = 0
 ReqEmailSince = 0
 GeoLocking = 0
-StrictVersionCheck = 1
+StrictVersionCheck = 0
```
Comment (mserajnik) [inline src/realmd/realmd.conf.dist.in:180]: And here.

File: src/game/Chat/Chat.cpp [hunk 15]
```diff
@@ -1863,10 +1864,10 @@ void ChatHandler::ExecuteCommand(char const* text)
...
         case CHAT_COMMAND_OK:
         {
-            if (!isAvailable(*command))
-            {
-                SendSysMessage(LANG_COMMAND_UNAVAILABLE);
-                return;
+            if (!isAvailable(*command))
+            {
+                SendSysMessage(LANG_COMMAND_UNAVAILABLE);
+                return;
```
Comment (mserajnik) [inline src/game/Chat/Chat.cpp:1870]: You aren't actually changing anything here.

File: src/game/Chat/Chat.cpp [hunk 16]
```diff
@@ -1470,28 +1471,28 @@ AccountTypes ChatHandler::GetAccessLevel() const
...
+    {
+        auto itrGranted = m_rbacAccountGrantedPermissions.find(m_session->GetAccountId());
+        if (itrGranted != m_rbacAccountGrantedPermissions.end())
+        {
+            if (cmd.PermissionMask & itrGranted->second)
+                return true;
+        }
+
+        return false;
+    }
```
Comment (mserajnik) [inline src/game/Chat/Chat.cpp:1495]: You aren't actually changing anything here either.

File: src/game/Chat/Chat.cpp [hunk 17]
```diff
@@ -1323,119 +1324,119 @@ ChatCommand * ChatHandler::getCommandTable()
...
+        }
+        case CHAT_COMMAND_UNKNOWN:
+        {
+            sLog.Out(LOG_DBERROR, LOG_LVL_ERROR, "Table `command` have nonexistent command '%s', skip.", cmdName.c_str());
+            return false;
+        }
+    }
+
+    return false;
+}
```
Comment (mserajnik) [inline src/game/Chat/Chat.cpp:1439]: You aren't actually changing anything here either.

File: src/game/Chat/Chat.cpp [hunk 18]
```diff
@@ -93,22 +93,23 @@ ChatCommand * ChatHandler::getCommandTable()
 
     static ChatCommand partyBotCommandTable[] =
     {
-        { "add",        SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotAddCommand,         "", nullptr },
-        { "clone",      SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotCloneCommand,       "", nullptr },
+        { "add",        SEC_PLAYER,      false, &ChatHandler::HandlePartyBotAddCommand,         "", nullptr },
+        { "clone",      SEC_PLAYER,      false, &ChatHandler::HandlePartyBotCloneCommand,       "", nullptr },
```
Comment (mserajnik) [inline src/game/Chat/Chat.cpp:97]: Why are you changing the permission from `SEC_ADMINISTRATOR` to `SEC_PLAYER` here? That doesn't seem right.

File: src/game/Chat/Chat.cpp [hunk 19]
```diff
@@ -93,22 +93,23 @@ ChatCommand * ChatHandler::getCommandTable()
 
     static ChatCommand partyBotCommandTable[] =
     {
-        { "add",        SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotAddCommand,         "", nullptr },
-        { "clone",      SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotCloneCommand,       "", nullptr },
+        { "add",        SEC_PLAYER,      false, &ChatHandler::HandlePartyBotAddCommand,         "", nullptr },
+        { "clone",      SEC_PLAYER,      false, &ChatHandler::HandlePartyBotCloneCommand,       "", nullptr },
         { "load",       SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotLoadCommand,        "", nullptr },
-        { "setrole",    SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotSetRoleCommand,     "", nullptr },
+        { "setrole",    SEC_PLAYER,      false, &ChatHandler::HandlePartyBotSetRoleCommand,     "", nullptr },
```
Comment (mserajnik) [inline src/game/Chat/Chat.cpp:99]: Changing permissions (see above).

File: src/game/Chat/Chat.cpp [hunk 20]
```diff
@@ -93,22 +93,23 @@ ChatCommand * ChatHandler::getCommandTable()
...
-        { "clone",      SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotCloneCommand,       "", nullptr },
+        { "add",        SEC_PLAYER,      false, &ChatHandler::HandlePartyBotAddCommand,         "", nullptr },
+        { "clone",      SEC_PLAYER,      false, &ChatHandler::HandlePartyBotCloneCommand,       "", nullptr },
         { "load",       SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotLoadCommand,        "", nullptr },
-        { "setrole",    SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotSetRoleCommand,     "", nullptr },
+        { "setrole",    SEC_PLAYER,      false, &ChatHandler::HandlePartyBotSetRoleCommand,     "", nullptr },
         { "attackstart",SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotAttackStartCommand, "", nullptr },
         { "attackstop", SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotAttackStopCommand,  "", nullptr },
-        { "aoe",        SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotAoECommand,         "", nullptr },
+        { "aoe",        SEC_PLAYER,      false, &ChatHandler::HandlePartyBotAoECommand,         "", nullptr },
```
Comment (mserajnik) [inline src/game/Chat/Chat.cpp:102]: Changing permissions (see above).

File: src/game/Chat/Chat.cpp [hunk 21]
```diff
@@ -93,22 +93,23 @@ ChatCommand * ChatHandler::getCommandTable()
...
-        { "cometome",   SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotComeToMeCommand,    "", nullptr },
-        { "usegobject", SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotUseGObjectCommand,  "", nullptr },
-        { "pause",      SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotPauseCommand,       "", nullptr },
-        { "unpause",    SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotUnpauseCommand,     "", nullptr },
+        { "focusmark",  SEC_PLAYER,      false, &ChatHandler::HandlePartyBotFocusMarkCommand,   "", nullptr },
+        { "clearmarks", SEC_PLAYER,      false, &ChatHandler::HandlePartyBotClearMarksCommand,  "", nullptr },
+        { "cometome",   SEC_PLAYER,      false, &ChatHandler::HandlePartyBotComeToMeCommand,    "", nullptr },
+        { "usegobject", SEC_PLAYER,      false, &ChatHandler::HandlePartyBotUseGObjectCommand,  "", nullptr },
+        { "pause",      SEC_PLAYER,      false, &ChatHandler::HandlePartyBotPauseCommand,       "", nullptr },
+        { "unpause",    SEC_PLAYER,      false, &ChatHandler::HandlePartyBotUnpauseCommand,     "", nullptr },
```
Comment (mserajnik) [inline src/game/Chat/Chat.cpp:109]: Changing permissions (see above).

File: src/game/Chat/Chat.cpp [hunk 22]
```diff
@@ -93,22 +93,23 @@ ChatCommand * ChatHandler::getCommandTable()
...
         { "unequip",    SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotUnequipCommand,     "", nullptr },
-        { "remove",     SEC_ADMINISTRATOR,      false, &ChatHandler::HandlePartyBotRemoveCommand,      "", nullptr },
+        { "remove",     SEC_PLAYER,      false, &ChatHandler::HandlePartyBotRemoveCommand,      "", nullptr },
```
Comment (mserajnik) [inline src/game/Chat/Chat.cpp:111]: Changing permissions (see above).

File: src/game/Chat/Chat.cpp [hunk 23]
```diff
@@ -422,7 +423,7 @@ ChatCommand * ChatHandler::getCommandTable()
         { "turn",           SEC_GAMEMASTER,     false, &ChatHandler::HandleGameObjectTurnCommand,      "", nullptr },
         { "info",           SEC_TICKETMASTER,   false, &ChatHandler::HandleGameObjectInfoCommand,      "", nullptr },
         { "ufinfo",         SEC_DEVELOPER,      false, &ChatHandler::HandleGameObjectUpdateFieldsInfoCommand, "", nullptr },
-        { "select",         SEC_TICKETMASTER,   false, &ChatHandler::HandleGameObjectSelectCommand,    "", nullptr },
+        { "select",         SEC_PLAYER,   false, &ChatHandler::HandleGameObjectSelectCommand,    "", nullptr },
```
Comment (mserajnik) [inline src/game/Chat/Chat.cpp:426]: Changing permissions (see above).

File: src/game/Chat/Chat.h [hunk 24]
```diff
@@ -282,6 +282,7 @@ class ChatHandler
         bool HandlePartyBotUnpauseCommand(char * args);
         bool HandlePartyBotUnequipCommand(char * args);
         bool HandlePartyBotRemoveCommand(char * args);
+        bool HandlePartyBotRepairitemsCommand(char * args);
```
Comment (mserajnik) [inline src/game/Chat/Chat.h:285]: It should probably be called `HandlePartyBotRepairItemsCommand` instead (the `I` of `Items` should be uppercased).

**Review by mserajnik — CHANGES_REQUESTED** [review]
This PR is a bit of a mess; you are changing a lot of things you shouldn't, including default configuration values. I am sure that's not intended; my reviews highlight those changes (I didn't look at the actual proposed command addition).

**Comment by ratkosrb** [comment]
These changes are unwelcome.

**Comment by gajet5** [comment]
Right. I don't understand why these changes were added to the PR.

**Comment by gajet5** [comment]
ok, I get it, my fault is that I didn't create a branch for my pr.

**Pull request closed without merging**